processor/src/errors.rs

```diff
@@ -478,6 +500,29 @@ pub enum AceError {
     #[error("num of wires must be less than 2^30 but was {0}")]
     TooManyWires(u64),
 }
+// DEBUG AND TRACE ERRORS
+// ================================================================================================
+
+/// Boxed error type for debug handlers.
+///
+/// Any error produced by a host's `DebugHandler` is converted into this boxed trait object.
+/// The alias documents the shape the VM expects; concrete host error types are allowed and
+/// are boxed by the VM at the call boundary.
+type DebugError = Box<dyn Error + Send + Sync + 'static>;
+
+/// Boxed error type for trace handlers.
+///
+/// Any error produced by a host's trace hook is converted into this boxed trait object.
+/// As with `DebugError`, concrete host error types are permitted and boxed at the boundary.
+type TraceError = Box<dyn Error + Send + Sync + 'static>;
+
+/// Boxed assertion context attached to failed assertions.
+///
+/// Hosts may return any type implementing `fmt::Debug + Send + Sync + 'static` from
+/// `BaseHost::on_assert_failed`. The VM then boxes it into this trait object and includes it in
+/// diagnostics. Prefer a concrete, sized type for `BaseHost::AssertInfo` (e.g., `Felt` or `()`),
+/// and let the VM perform the boxing to avoid double-boxing.
+pub(crate) type AssertInfo = Box<dyn fmt::Debug + Send + Sync + 'static>;
```

Comment:
I would've expected this to be `fmt::Display`. `Debug` is often automatically derived, while for error messages, you usually want a custom impl. I think this is in line with what Rust docs suggest (https://doc.rust-lang.org/std/fmt/trait.Display.html):
> Display is similar to Debug, but Display is for user-facing output, and so cannot be derived.